src/components/Review/ReviewTable.jsx

```diff
@@ -30,27 +27,28 @@ class ReviewTable extends React.Component {
     };
   }
 
-  componentDidUpdate(prevProps) {
-    const { survey, data } = this.props;
-    if (prevProps.survey !== survey) {
-      const newCheckboxArray = data.reduce((acc, curr) => {
-        acc[curr.id] = false;
-        return acc;
-      }, {});
-      this.setState({ checkboxArray: newCheckboxArray, checkAll: false });
+  getCheckAllValue(checkboxArray, pagination){
+    const lastIndexOnPage = pagination.size * pagination.page > checkboxArray.length ? 
+    checkboxArray.length : 
+    pagination.size *  pagination.page;
+
+    for (let i = pagination.size * (pagination.page - 1); i < lastIndexOnPage; i++ ){
+      if(checkboxArray[i].isChecked === false){
+        return false;
+      }
     }
+
+    return true;
```

Comment:
Same code (with a nice review comment 😉 ) in another component, could be exported /  imported